queries/sun_position.py

```diff
@@ -0,0 +1,623 @@
+"""
+
+    Greynir: Natural language processing for Icelandic
+
+    Example of a grammar query processor module.
+
+    Copyright (C) 2021 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    This module handles queries regarding time of sunrise/sunset.
+
+"""
+from typing import Dict, List, Tuple, Optional, Union, cast
+from tree import Result, Node
+from query import Query, QueryStateDict
+from queries import AnswerTuple, LatLonTuple, MONTH_ABBREV, gen_answer
+
+import datetime
+import random
+import re
+import requests
+from bs4 import BeautifulSoup  # type: ignore
+from cachetools import cached, TTLCache
+from settings import changedlocale
+from geo import (
+    distance,
+    in_iceland,
+    capitalize_placename,
+    ICE_PLACENAME_BLACKLIST,
+)
+from iceaddr import placename_lookup  # type:ignore
+
+
+# Indicate that this module wants to handle parse trees for queries,
+# as opposed to simple literal text strings
+HANDLE_TREE = True
+
+# The grammar nonterminals this module wants to handle
+QUERY_NONTERMINALS = {"QSunQuery"}
+
+_SUN_QTYPE = "SunPosition"
+
+TOPIC_LEMMAS = [
```

Comment:
FYI, lemmas can also include verbs, such as 'setjast', 'hníga' and 'rísa'

Reply:
Good point, added a few verbs!